lib/engine/game/g_1840/step/buy_train.rb

```diff
@@ -47,6 +47,10 @@ def process_scrap_train(action)
             @game.scrap_train(action.train, action.entity)
           end
 
+          def scrap_trains_button_only?
```

Comment:
Remove from here and other games as per the comment in the scrap_trains view file